# How many units of Galxay 8 to produce?

This project is completed by Asyl Uraimova for the Quantitative Analytics course at SKK GSB

In [6]:
import numpy as np
from scipy import optimize

The potential demand is given in million USD below

In [2]:
demand=[2.2,2.4,2.6,2.8,3,3.2,3.4,3.6]
demand

[2.2, 2.4, 2.6, 2.8, 3, 3.2, 3.4, 3.6]

The potential losses are given in million USD below

In [7]:
p=list(np.array([5, 10, 15, 20, 20, 15, 10, 5])/100)
p

[0.05, 0.1, 0.15, 0.2, 0.2, 0.15, 0.1, 0.05]

# The objective function

It is given that the cost of lost sales is 500USD and the disposal cost is 300USD.

In [10]:
def expected_loss(x):
    y=0
    for i in range(len(p)):
        #print(demand[i], p[i], max(demand[i]-x,0), max(x-demand[i],0)
        y=y+(500*max(demand[i]-x, 0)+300*max(x-demand[i],0))*p[i]
    return y

# Optimizing production

We need to find the optimal production quantity that will bring the most sales while minimizing its potential losses. 

In [11]:
optimize.minimize(expected_loss, x0=np.array([2.2]), bounds=[(2.2,3.6)] )

      fun: array([110.00000007])
 hess_inv: <1x1 LbfgsInvHessProduct with dtype=float64>
      jac: array([49.32269491])
  message: b'CONVERGENCE: REL_REDUCTION_OF_F_<=_FACTR*EPSMCH'
     nfev: 110
      nit: 7
   status: 0
  success: True
        x: array([3.])

We found that at a production level 3 million units Samsung will optimize minimize its potential loss that equals 110 million USD. 